p2p/net/conn.go

```diff
@@ -140,101 +152,111 @@ func (c *FormattedConnection) publish(message []byte) {
 	c.networker.EnqueueMessage(IncomingMessageEvent{c, message})
 }
 
-// incomingChannel returns the incoming messages channel
-func (c *FormattedConnection) incomingChannel() chan []byte {
-	return c.formatter.In()
-}
-
 // Send binary data to a connection
 // data is copied over so caller can get rid of the data
 // Concurrency: can be called from any go routine
 func (c *FormattedConnection) Send(m []byte) error {
-	err := c.formatter.Out(m)
+	c.wmtx.Lock()
+	_, err := c.w.WriteRecord(m)
 	if err != nil {
+		c.wmtx.Unlock()
```

Comment:
minor comment - consider calling `Unlock` immediately after `WriteRecord` instead of twice afterward